### PR TITLE
Rename Merkle links key from mlink to link

### PR DESCRIPTION
**Pull request by mildred**
Merges to #37

Replaces `mlink` occurrences with `link` with a paragraph to explain how merkle-links are formed. [Original comment](/ipfs/specs/pull/37#discussion_r45558988) and [follow up](/ipfs/specs/pull/37#issuecomment-169775469)

### CODE REVIEW

**Comment by jbenet** [comment]
this LGTM. though do we want `link` or `@link` to signify importance, allow people to use `link` for other things (and `\@link` if they need to).

**Comment by mildred** [comment]
I didn't understand what you said about `@link` and what does it mean to have an important link.

**Comment by jbenet** [comment]
Sorry, let me rephrase: do we want `link` or `@link`? the `@` helps signify importance, as it is a clearly distinguished value, and we can do the directives + escaping. this also allows people to use the value `link` themselves without creating links accidentally. 

(i myself am leaning towards directives + escaping)

**Comment by mildred** [comment]
If you want to add escaping, we can either let the application do the escaping on their own, or handle the escaping ourselves. Escaping implies that we will be able to provide the application with either the escaped (original) version, or the unescaped (with link information removed) objects. If the application wants to look at links, it will not be able to do so in the unescaped version the `@link` property will be removed). The application will also have to specify which version it wants and this may create confusion. *"Which version should I get?"*

I'd be tempted to say it doesn't matter if the application create links unknowingly. We'll just never use them. So let's use `link`, that's fine.

If we want to provide an easy way to avoid that, just use `@links` but let the application itself do the escaping. In any case, we probably don't want to bother with escaping in IPLD core. We'll just provide a library that the application would be free to use or not.

**Comment by jbenet** [comment]
Very well thought out. thanks

> If we want to provide an easy way to avoid that, just use @links but let the application itself do the escaping. In any case, we probably don't want to bother with escaping in IPLD core. We'll just provide a library that the application would be free to use or not.

Yep totally agreed here, i think this ^ is the right choice.

**Comment by jbenet** [comment]
thanks @mildred
